### PR TITLE
Burnes/ts update

### PR DESCRIPTION
**Pull request by burnes**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## Unreleased
+
+### Development
+
+- Use Typescript 4.0.3
+- Adapted a lot of dependecies to align with lovely-ra
```
Comment (natterstefan) [inline CHANGELOG.md:8]: Does https://github.com/lovelysystems/lovely-ra/pull/66#discussion_r502284245 fix the changes here too?

**Review by natterstefan — COMMENTED** [review]
See my comment, please.

File: CHANGELOG.md (hunk 1 above)
Comment (burnes) [reply CHANGELOG.md:8]: probably yes, I'll add the config in this project too, as it is still missing, and I also have to fix the tests here, as they fail on the individual checkout.

File: package.json [hunk 2]
```diff
@@ -19,6 +19,9 @@
     "build:cjs": "rimraf ./build && tsc --outdir build",
     "build:esm": "rimraf ./build/esm && tsc --outDir build/esm --module es2015",
     "build:copy-files": "node ./scripts/copy-files.js",
+    "lint": "eslint --fix 'src/**/*.{js,jsx,ts,tsx}'",
+    "lint-staged": "yarn lint-js-staged",
+    "lint-js-staged": "git diff --diff-filter=d --cached --name-only -z -- '*.js' '*.jsx' | xargs -0 -I % sh -c 'git show \":%\" | eslint --stdin --stdin-filename \"%\";'",
```
Comment (natterstefan) [inline package.json:24]: There are no githooks, so we could remove these two. Or is travis executing them?

**Review by natterstefan — APPROVED** [review]

**Pull request merged**